python/tests/test_benchmark.py

```diff
@@ -33,17 +33,22 @@ def get_sgnn_res(
 
 
 def assert_knn_equal(
-    knn_df: DataFrame, id_col_name: str, distances: np.ndarray, indices: np.ndarray
+    knn_df: DataFrame,
+    id_col_name: str,
+    distances: np.ndarray,
+    indices: np.ndarray,
+    total_tol: float = 1e-3,
 ) -> None:
     res_pd: pd.DataFrame = knn_df.sort(f"query_{id_col_name}").toPandas()
     mg_indices: np.ndarray = np.array(res_pd["indices"].to_list())
     mg_distances: np.ndarray = np.array(res_pd["distances"].to_list())
 
-    assert array_equal(mg_indices, indices)
     assert array_equal(mg_distances, distances)
 
+    # set total_tol because two nearest neighbors may have the same distance to the query
+    assert array_equal(mg_indices, indices, total_tol=total_tol)
```

Comment:
In that case, if indices differ should check that the distances are the same.   This tol could let a bug slip through.

Reply:
Created a follow-up PR to check the returned distances. The checking seemed redundant due to the line "assert array_equal(mg_distances, distances)" which already gets returned distances checked. Perhaps we should recalculate the distance according to indices and then check whether the recalculated distances are equal. Quite troublesome if so but let me know if your comments meant something else.

Reply:
Ah ok.   I agree that the returned minimum distances matching is a good check and may be sufficient.   Is there then a need to check the indices at all?   Do we have confidence that the distances are correct for the corresponding indices?